go.mod

```
@@ -1,6 +1,8 @@
 module github.com/rancher/rke2
 
-go 1.19
+go 1.21
+
+toolchain go1.21.7
```

```suggestion
go 1.19
```

k3s is still on `go 1.19` for this branch.

Remove the `toolchain` line and then run the tidy with `GOTOOLCHAIN=$(. ./scripts/version.sh &>/dev/null; echo $VERSION_GOLANG) go mod tidy` and it will prevent go1.22 from adding this unwanted option. We're still building with golang <1.22 so there is no need for 1.22 specific changes on this branch.